test/unittest/memory/unittest_cache_loader.cpp

```diff
@@ -221,13 +223,14 @@ TEST_F(CacheLoaderTest, load_async_02_p) {
   };
 
   int id = loader->loadAsync(1, complete1);
+  future.wait();
+  pool->flush();
+
```

Comment:
In synchronous loading tests, it does not check if `mem->getAddr()` is a nullptr after load.
Instead, it checks if the memory is properly loaded after load.

https://github.com/nnstreamer/nntrainer/blob/8104cbe29bce20019bb829edee5dd5d050e5498a/test/unittest/memory/unittest_cache_loader.cpp#L118-L122


On the other hand, asynchronous loading tests check if memory is properly loaded with execution order by using a lambda function `complete`. Current load_async test logic checks if all memory is swapped out after `loader->loadAsync()`. While no flush logic is found in `loadAsync()` (at least on my side), checking if the memory data is swapped out after loading and before the cache pool flushes seems incorrect. Therefore, I added `pool->flush()` before checking if the memory address is a nullptr.